lib/tenji_maker.rb

```diff
@@ -1,43 +1,38 @@
+# frozen_string_literal: true
+
+TENJI_3BIT_TABLE = {
+  '0' => '---',
+  '1' => '--o',
+  '2' => '-o-',
+  '3' => '-oo',
+  '4' => 'o--',
+  '5' => 'o-o',
+  '6' => 'oo-',
+  '7' => 'ooo'
+}.freeze
+
+# 母音と子音を3bit単位で判別するために、点の有無を「① ④ ② ⑤ ③ ⑥ 」の順に列指向で並べている
 class TenjiMaker
   def to_tenji(text)
-    # 以下はサンプルの仮実装なので、このcase文は全部消して自作ロジックに書き直すこと
-    case text
-    when 'A HI RU'
-      <<~TENJI.chomp
-        o- o- oo
-        -- o- -o
-        -- oo --
-      TENJI
-    when 'KI RI N'
-      <<~TENJI.chomp
-        o- o- --
-        o- oo -o
-        -o -- oo
-      TENJI
-    when 'SI MA U MA'
-      <<~TENJI.chomp
-        o- o- oo o-
-        oo -o -- -o
-        -o oo -- oo
-      TENJI
-    when 'NI WA TO RI'
-      <<~TENJI.chomp
-        o- -- -o o-
-        o- -- oo oo
-        o- o- o- --
-      TENJI
-    when 'HI YO KO'
-      <<~TENJI.chomp
-        o- -o -o
-        o- -o o-
-        oo o- -o
-      TENJI
-    when 'KI TU NE'
-      <<~TENJI.chomp
-        o- oo oo
-        o- -o o-
-        -o o- o-
-      TENJI
+    tenji_serials = text.split(' ').map { |romaji| make_tenji(romaji) }
+    [0..1, 2..3, 4..5].map do |range|
+      tenji_serials.map { |tenji_serial| tenji_serial[range] }.join(' ')
+    end.join("\n")
+  end
+
+  def make_tenji(romaji)
+    if romaji.length == 1
+      case romaji
+      when 'N'     ; '---ooo'
+      else         ; "#{make_tenji_3bit('xxxOAIUE', romaji)}---"
+      end
+    else
+      case romaji
+      when /^[YW]/ ; "#{make_tenji_3bit('WxYxxxxx', romaji[0])}#{make_tenji_3bit('xxAUxxOx', romaji[1])}"
+      else         ; "#{make_tenji_3bit('xxxOAIUE', romaji[1])}#{make_tenji_3bit('xKNHRSTM', romaji[0])}"
+      end
     end
   end
+
+  def make_tenji_3bit(target, char) = (target =~ /#{char}/).to_s.sub(/./, TENJI_3BIT_TABLE)
```

Comment:
`TENJI_3BIT_TABLE`は配列にした上で、こんなふうにするとよりシンプルですね。

```suggestion
  def make_tenji_3bit(target, char) = TENJI_3BIT_TABLE[target.index(char)]
```

Reply:
たしかに、配列にするとシンプルに書けますね。。
ご提案ありがとうございます〜。